app/src/util/market_creation_status_data.ts

```diff
@@ -0,0 +1,105 @@
+enum Types {
+  Ready = 'Ready',
+  PostingQuestion = 'Posting question to realitio',
+  PrepareCondition = 'Prepare condition',
+  ApprovingCollateral = 'Approving collateral',
+  CreateMarketMaker = 'Create market maker',
+  ApproveCollateralForMarketMaker = 'Approve collateral for market maker',
+  AddFunding = 'Add funding in market maker',
+  InitialTradeInMarketMaker = 'initial trade in market maker',
+  Done = 'Done',
+  Error = 'Error',
+}
+
+interface Ready {
+  readonly _type: Types.Ready
+}
+interface PostingQuestion {
+  readonly _type: Types.PostingQuestion
+}
+interface PrepareCondition {
+  readonly _type: Types.PrepareCondition
+}
+interface ApprovingCollateral {
+  readonly _type: Types.ApprovingCollateral
+}
+interface CreateMarketMaker {
+  readonly _type: Types.CreateMarketMaker
+}
+interface ApproveCollateralForMarketMaker {
+  readonly _type: Types.ApproveCollateralForMarketMaker
+}
+interface AddFunding {
+  readonly _type: Types.AddFunding
+}
+interface InitialTradeInMarketMaker {
+  readonly _type: Types.InitialTradeInMarketMaker
+}
+interface Done {
+  readonly _type: Types.Done
+}
+interface Error {
+  readonly _type: Types.Error
+  readonly error: Error
+}
+
+export type MarketCreationStatusType =
```

Comment:
Maybe we can use the same name for the type and the helper object, as we do with `RemoteData`. That way you only need to import one name from this module.

Reply:
Also drop the `Data` suffix.

Reply:
:+1: